### PR TITLE
NF: assert->Assert

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
As timber state, this is never activated. So if we want it, we must uses Assert.

Or we could delete it, the app worked fine up to now without them

### CODE REVIEW

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
`Assert` throws an error. 

I don't feel that crashing is the correct move and we'd be better throwing a class deriving from `RuntimeException`

I especially don't think we should be using `AssertionError`, as it'd be perfectly valid for the test framework to perform some static work in the constructor which would cause test failures in the future.

File: AnkiDroid/src/main/java/com/ichi2/libanki/Card.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import android.database.Cursor;
 import android.text.TextUtils;
 
+import com.ichi2.anki.BuildConfig;
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Card.java:25]: BuildConfig?

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Card.java:25]: Timber suggest to use BuildConfig to check whether it's debug mode and fail in this case.

I tried it, but felt like that was not what we were looking for either. Of course, this does not means that this import should be kept.

I don't understand what you want. either we throw a RuntimeException or we don't crash. But we can't do both simultaneously.
I'm quite fine with any other solution. Just seems that this is not what we are looking for

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Card.java:25]: We can throw a class deriving from `RuntimeException` and there's a good chance that it's caught.

Throwing an `Error` means that it should be explicitly handled, and very few places in the code do this.

Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Card.java:25]: Actually, I don't understand why you'd want to catch it, unless we try to correct it. Those are things that are not supposed to be possible in the first place. If we catch it, either we send it again and I don't see the point, or it's ignored and that's worse.
We want to be warned if this ever occurs

**Comment by david-allison-1** [comment]
Probably blocked by: https://github.com/ankidroid/Anki-Android/issues/7747

File: AnkiDroid/src/main/java/com/ichi2/libanki/Card.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Card.java:25]: I removed BuildConfig via an interactive rebase while checking this locally

**Review by mikehardy — COMMENTED** [review]
I do not like the idea of a previously silent 'assert' now turning into an "Assert" crash without recourse for the user, or any new RuntimeException doing similar.

The right way to solve this is either:

1- continue ignoring it, and if Timber complains, actually remove the assert calls
2- turn each one into a "sendExceptionReport/replace-value-reasonable-default" stanza
3- turn each one into a "your database is corrupt / run check database" stanza and add the appropriate checks / reasonable default repairs to check database

Of the 3, 1 is obviously easiest and I'm going to close this so it is also the default stance

3 is the most correct as these are database corruption issues, and indicate a database needs to be repaired.

**Pull request closed without merging**